src/background/deinflect.test.ts

```diff
@@ -309,7 +315,150 @@ describe('deinflect', () => {
+      if (secondary.length) {
+        reasonChains[1] = [...secondary];
+      }
+      expect(match!.reasonChains).toEqual(reasonChains);
+    }
+  });
+
+  it('deinflects なさる as respectful alternative for する', () => {
+    const cases: Array<[string, string, Array<Array<Reason>>]> = [
+      [
+        'なさい',
+        'する',
+        [[Reason.Respectful, Reason.Imperative], [Reason.MasuStem]],
+      ],
+      ['仕事なさる', '仕事', [[Reason.SuruNoun, Reason.Respectful]]],
+      [
+        'エンジョイなさって',
+        'エンジョイ',
+        [[Reason.SuruNoun, Reason.Respectful, Reason.Te]],
+      ],
+      ['喜びなさった', '喜ぶ', [[Reason.Respectful, Reason.Past]]],
+    ];
+
+    for (const [inflected, plain, reasonChains] of cases) {
+      const result = deinflect(inflected);
+      const match = result.find((candidate) => candidate.word == plain);
+      expect(match).toBeDefined();
+      expect(match!.reasonChains).toEqual(reasonChains);
+    }
+
+    // // masu-stem + nasai should still be recognized as -nasai form
+    // const result = deinflect('喜びなさい');
+    // const match = result.find((candidate) => candidate.word == '喜ぶ');
+    // expect(match).toBeDefined();
+    // expect(match!.reasonChains).toEqual([[Reason.Nasai]]);
+  });
+
+  it('deinflects になる as respectful speech', () => {
+    const cases: Array<[string, string, Array<Reason>]> = [
+      ['到着になります', '到着', [Reason.SuruNoun]],
+      ['読みになります', '読む', []],
+      ['見えになります', '見える', []],
+    ];
+
+    for (const [inflected, plain, reasonAddition] of cases) {
+      const result = deinflect(inflected);
+      const match = result.find((candidate) => candidate.word == plain);
+      expect(match).toBeDefined();
+      expect(match!.reasonChains).toEqual([
+        [...reasonAddition, Reason.Respectful, Reason.Polite],
+      ]);
+    }
+  });
+
+  it('deinflects humble or dialectal continuous forms', () => {
+    const cases: Array<[string, string, Array<Reason>]> = [
+      ['行っておる', '行く', []],
+      ['行っており', '行く', [Reason.MasuStem]],
+      ['行っとる', '行く', []],
+      ['行っとり', '行く', [Reason.MasuStem]],
+      ['読んでおる', '読む', []],
+      ['読んでおり', '読む', [Reason.MasuStem]],
+      ['読んどる', '読む', []],
+      ['読んどり', '読む', [Reason.MasuStem]],
+      ['起きておる', '起きる', []],
+      ['起きており', '起きる', [Reason.MasuStem]],
+      ['起きとる', '起きる', []],
+      ['起きとり', '起きる', [Reason.MasuStem]],
+    ];
+
+    for (const [inflected, plain, reason] of cases) {
+      const result = deinflect(inflected);
+      const match = result.find((candidate) => candidate.word == plain);
+      expect(match).toBeDefined();
+      expect(match!.reasonChains).toEqual([
+        [Reason.HumbleOrDialectal, Reason.Continuous, ...reason],
+      ]);
+    }
+  });
+
+  it('deinflects 致す as humble speech for する', () => {
+    const cases: Array<[string, string, Array<Array<Reason>>]> = [
+      ['いたす', 'する', [[Reason.Humble]]],
+      ['致す', 'する', [[Reason.Humble]]],
```

Comment:
Ideally we wouldn't inflect these words when they appear by themselves since they have their own dictionary entries.